src/lib/clk.c

```diff
@@ -115,6 +115,19 @@ uint64_t clock_ms_to_ticks(int clock, uint64_t ms)
 	return ticks;
 }
 
+uint64_t clock_ticks_per_sample(int clock, uint32_t sample_rate)
+{
+	struct clock_info *clk_info = clocks_get() + clock;
+	uint32_t ticks_per_msec =
+		clk_info->freqs[clk_info->current_freq_idx].ticks_per_msec;
+	uint64_t ticks_per_sample = ticks_per_msec * 1000 /
```

Comment:
sample_rate is mandatory here. If it's not passed in then return 0

Reply:
> sample_rate is mandatory here. If it's not passed in then return 0

Fair, let me change.